quickwit/quickwit-config/src/source_config/mod.rs

```diff
@@ -79,6 +79,9 @@ pub struct SourceConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[serde(rename = "transform")]
     pub transform_config: Option<TransformConfig>,
+
+    // Denotes if the data is in json or any other format.
+    pub is_json_format: bool,
```

Comment:
Let's use an `data_format` (or `input_format`?) enum for this parameter that defaults to JSON:
```rust
#[derive(Debug, Clone, Copy, Default, ...)]
enum SourceDataFormat {
  #[default]
  Json,
  #[serde(alias = "plain")]
  PlainText, 
}
```